lib/g3-types/src/acl_set/dst_host.rs

```diff
@@ -16,39 +16,52 @@
 
 use crate::acl::{
     AclAction, AclChildDomainRule, AclChildDomainRuleBuilder, AclExactHostRule, AclNetworkRule,
-    AclNetworkRuleBuilder, AclRegexSetRule, AclRegexSetRuleBuilder,
+    AclNetworkRuleBuilder, AclRegexSetRule, AclRegexSetRuleBuilder, ActionContract,
 };
 use crate::net::Host;
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct AclDstHostRuleSetBuilder {
-    pub exact: Option<AclExactHostRule>,
-    pub child: Option<AclChildDomainRuleBuilder>,
-    pub regex: Option<AclRegexSetRuleBuilder>,
-    pub subnet: Option<AclNetworkRuleBuilder>,
+pub struct AclDstHostRuleSetBuilder<Action = AclAction> {
+    pub exact: Option<AclExactHostRule<Action>>,
+    pub child: Option<AclChildDomainRuleBuilder<Action>>,
+    pub regex: Option<AclRegexSetRuleBuilder<Action>>,
+    pub subnet: Option<AclNetworkRuleBuilder<Action>>,
+    pub missing_action: Option<Action>,
 }
 
-impl AclDstHostRuleSetBuilder {
-    pub fn build(&self) -> AclDstHostRuleSet {
-        let mut missed_action = AclAction::Permit;
+impl<Action> Default for AclDstHostRuleSetBuilder<Action> {
+    fn default() -> Self {
+        Self {
+            exact: None,
+            child: None,
+            regex: None,
+            subnet: None,
+            missing_action: None,
+        }
+    }
+}
+
+impl<Action: ActionContract> AclDstHostRuleSetBuilder<Action> {
+    pub fn build(&self) -> AclDstHostRuleSet<Action> {
+        let mut missed_action = self.missing_action.unwrap_or_else(Action::default_permit);
 
         let exact_rule = self.exact.as_ref().map(|rule| {
-            missed_action = missed_action.restrict(rule.missed_action());
+            missed_action = rule.missed_action().max(missed_action);
```

Comment:
The concern here is that, wr can not say `intercept` is more restrict than `detour`. A different build approach may be needed for `AclAction` and `ProtocolInspectAction`. You can squash the commits into smaller ones, and I will the build logic here after merged.

Reply:
I agree. I'll get to the squashing now so you can take over.

Note that besides your feedback regarding the `restrict` thing and also the builder clone that you can probably easily fix yourself, for which thank you, there is also the more severe issue, which is the bug I was talking to you about in the issue. Regarding also blocking non-ws traffic once the ws policy is defined. So that's something to look out for as well.